huntsman/dome/musca.py

```diff
@@ -63,24 +63,33 @@ def __init__(self, *args, **kwargs):
 
     @property
     def is_open(self):
-        v = self._get_shutter_status_dict()[Protocol.SHUTTER]
+        v = self.status[Protocol.SHUTTER]
         return v == Protocol.OPEN
 
     @property
     def is_closed(self):
-        v = self._get_shutter_status_dict()[Protocol.SHUTTER]
+        v = self.status[Protocol.SHUTTER]
         return v == Protocol.CLOSED
 
     @property
     def door_open(self):
-        v = self._get_shutter_status_dict()[Protocol.DOOR]
+        v = self.status[Protocol.DOOR]
         return v == Protocol.DOOR_OPEN
 
     @property
     def door_closed(self):
-        v = self._get_shutter_status_dict()[Protocol.DOOR]
+        v = self.status[Protocol.DOOR]
         return v == Protocol.DOOR_CLOSED
 
+    @property
+    def status(self):
+        """A dictionary containing all status info for dome.
+
+        TODO: Add other info (e.g. dome moving)
+        TODO: Auto-update this every ~60s
+        """
+        return(self._get_shutter_status_dict())
```

Comment:
I think what we should do is set up a private variable `_status` as well as a timer, e.g. something like

```python
from pocs.utils import CoutdownTimer

...

def __init__(self):
    # other stuff
    self._status = dict()
    self._status_delay = 5 # seconds
    self._status_timer = CountdownTimer(self._status_delay)

def status(self):
    if self._status_timer.expired():
        self._status = self._get_shutter_status_dict()
        self._status_timer = CountdownTimer(self._status_delay)

    return self._status
```